include/groov/mmio_bus.hpp

```diff
@@ -58,28 +106,46 @@ template <typename HardwareInterface = cpp_mem_iface> struct mmio_bus {
     static auto write(auto addr, decltype(Mask) value) -> async::sender auto {
         using base_type = decltype(Mask);
 
-        constexpr std::size_t lsb = std::countr_zero(Mask);
-        constexpr std::size_t width = std::countr_one(Mask >> lsb);
-        constexpr bool byte_aligned = (lsb % 8) == 0;
-        constexpr bool contiguous = std::popcount(Mask) == width;
-        constexpr bool pow2_width = std::has_single_bit(width);
-        constexpr bool accessible_width = width & (8u | 16u | 32u | 64u);
-        constexpr bool subword_access_en =
-            byte_aligned && contiguous && pow2_width && accessible_width;
-
-        if constexpr (subword_access_en) {
-            using subword_t = detail::uint_of_width_t<width>;
-            constexpr auto byte_offset = lsb / 8;
-            auto const subword_addr = addr + byte_offset;
-            auto const subword_val = static_cast<subword_t>(value >> lsb);
+        using subwords_aligned =
+            detail::mp_filter_q<detail::aligned_for<iface>,
+                                detail::subword_permuations>;
+
+        using subwords_aligned_and_fit =
+            detail::mp_filter_q<detail::fits_within<base_type>,
+                                subwords_aligned>;
+
+        using subword_candidate_index =
+            detail::mp_find_if_q<subwords_aligned_and_fit,
+                                 detail::subword_satisfies<Mask, IdMask>>;
+
+        constexpr bool found_subword =
+            subword_candidate_index::value <
+            detail::mp_size<subwords_aligned_and_fit>::value;
```

Comment:
**thought**:  There are a couple of alternatives for this metaprogamming pattern - it's your choice, but here's what I've experienced:

1. `find_if` then `index != size` then `at`
2. `copy_if` then `not empty` then `first`

I often find the `copy_if` version easier.

- `find_if` springs to mind quicker because we'd use it at runtime for perf/short-circuit, but at compile time I don't know that applies
- `empty` is an easier terser check than `index < size` (especially because `bool_constant` has an implicit conversion to `bool`)
- likewise `mp_first` is terser than `mp_at`